index.d.ts

```diff
@@ -475,7 +475,7 @@ declare namespace Moleculer {
 		basePath?: string;
 	}
 
-	type ActionSchema<S = ServiceSettingSchema> = {
+	type ActionSchema<S extends ServiceSettingSchema = ServiceSettingSchema> = {
```

Comment:
My concern regarding the breaking change is the use of `extends ServiceSettingSchema` in all of these places.  That will break any consumers that are not currently extending that interface.  Can you try eliminating all of the `extends ServiceSettingSchema` from all locations and see if the other change to the implements clause in the `Service` class resolves the overall errors with TS?